src/org/labkey/test/components/react/QueryChartDialog.java

```diff
@@ -0,0 +1,330 @@
+    public QueryChartDialog(String title, WebDriver driver, QueryGrid grid)
+    {
+        super(new ModalDialogFinder(driver).withTitle(title));
+        _queryGrid = grid;
+    }
+
+    public QueryChartDialog setName(String value)
+    {
+        elementCache().nameInput.set(value);
+        return this;
+    }
+
+    public String getName()
+    {
+        return elementCache().nameInput.get();
+    }
+
+    public QueryChartDialog setShared(boolean checked)
+    {
+        elementCache().sharedCheckbox.set(checked);
+        return this;
+    }
+
+    public boolean getShared()
+    {
+        return elementCache().sharedCheckbox.get();
+    }
+
+    // field selects
+
+    /*
+        X axis is an option for bar, box, line, scatter charts
+     */
+    public QueryChartDialog selectXAxis(String field)
+    {
+        elementCache().reactSelectByLabel("X Axis").select(field);
+        return this;
+    }
+
+    public String getXAxisSelection()
+    {
+        return elementCache().reactSelectByLabel("X Axis").getValue();
+    }
+
+    /*
+        Y Axis is an option for bar, box, line, scatter charts
+     */
+    public QueryChartDialog selectYAxis(String field)
+    {
+        elementCache().reactSelectByLabel("Y Axis").select(field);
+        return this;
+    }
+
+    public String getYAxisSelection()
+    {
+        return elementCache().reactSelectByLabel("Y Axis").getValue();
+    }
+
+    /*
+        groupBy is an option for bar charts only
+     */
+    public QueryChartDialog selectGroupBy(String field)
+    {
+        elementCache().reactSelectByLabel("Group By").select(field);
+        return this;
+    }
+
+    /*
+     * color is an option for Box, Scatter charts
+     */
+    public QueryChartDialog selectColor(String field)
+    {
+        elementCache().reactSelectByLabel("Color").select(field);
+        return this;
+    }
+
+    /*
+        shape is an option for Scatter, Box charts
+     */
+    public QueryChartDialog selectShape(String field)
+    {
+        elementCache().reactSelectByLabel("Shape").select(field);
+        return this;
+    }
+
+    /*
+        series is an option for line
+     */
+    public QueryChartDialog selectSeries(String field)
+    {
+        elementCache().reactSelectByLabel("Series").select(field);
+        return this;
+    }
+
+    /*
+        categories is an option for pie charts
+     */
+    public QueryChartDialog selectCategories(String field)
+    {
+        elementCache().reactSelectByLabel("Categories").select(field);
+        return this;
+    }
+
+    // chart type selection
+    /*
+        Note: Chart Type is not settable when opened for edit/ it is only settable
+        when creating the chart
+     */
+    public QueryChartDialog setChartType(CHART_TYPE chartType)
+    {
+        elementCache().selectChartType(chartType);
+        return this;
+    }
+
+    public CHART_TYPE getSelectedChartType()
+    {
+        return elementCache().getSelectedChartType();
+    }
```

Comment:
```suggestion
    public QueryChartDialog setChartType(CHART_TYPE chartType)
    {

        if (getSelectedChartType().equals(chartType))
            return this;

        // note: selecting a chart type will cause most select fields to go stale
        var el = elementCache().chartBuilderType.withAttribute("data-name", chartType.getChartType())
                .waitForElement(this, 1500);
        getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(el));
        el.click();
        WebDriverWrapper.waitFor(()-> getSelectedChartType().equals(chartType),
                "The requested chart type did not become selected", 2000);

        return this;
    }

    public CHART_TYPE getSelectedChartType()
    {
        var selectedEl = elementCache().chartBuilderType.withAttributeContaining("class", "selected")
                .waitForElement(this, 1500);
        String dataName = selectedEl.getAttribute("data-name");
        return CHART_TYPE.fromChartType(dataName);
    }
```

Reply:
This would clean up the elementCache and remove the methods that aren't really elements on the dialog.